File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/date/DifferenceOfTimestampsTest.java
package com.fincity.nocode.kirun.engine.function.system.date;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonPrimitive;

import reactor.test.StepVerifier;

public class DifferenceOfTimestampsTest {

    DifferenceOfTimestamps dts = new DifferenceOfTimestamps();

    ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
        new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository());

    @Test
    void invalidTest(){

        fep.setArguments(Map.of("isoDateOne", new JsonPrimitive("2029-05-95T06:04:18.073Z"),
        "isoDateTwo", new JsonPrimitive("2029-05-05T06:04:18.073Z")));

        StepVerifier.create(dts.execute(fep))
                .expectError().verify();

        fep.setArguments(Map.of("isoDateOne", new JsonPrimitive("2029-05-05T06:04:18.073Z")));

        StepVerifier.create(dts.execute(fep))
                .expectError().verify();
    }

    @Test
	void test() {

		fep.setArguments(Map.of("isoDateOne", new JsonPrimitive("2024-09-13T23:52:34.633-05:30"),
        "isoDateTwo", new JsonPrimitive("2024-09-13T23:52:34.633Z")));

        StepVerifier.create(dts.execute(fep))
                .expectNextMatches(res -> res.next()
                .getResult()
                .get("result")
                .getAsInt() == -330)
                .verifyComplete();

	}

	@Test
	void test1() {

		fep.setArguments(Map.of("isoDateOne", new JsonPrimitive("2024-09-13T23:52:34.633-05:30"),
        "isoDateTwo", new JsonPrimitive("2024-09-12T23:52:34.633Z")));

        StepVerifier.create(dts.execute(fep))
            .expectNextMatches(res -> res.next()
            .getResult()
            .get("result")
            .getAsInt() == -1770)
            .verifyComplete();

	}

    @Test
    void test3() {

        fep.setArguments(Map.of("isoDateOne", new JsonPrimitive("2023-09-12T23:52:34.633Z"),
        "isoDateTwo", new JsonPrimitive("2024-09-13T23:52:34.633-05:30")));


        StepVerifier.create(dts.execute(fep))
            .expectNextMatches(res -> res.next()
            .getResult()
            .get("result")
            .getAsInt() == 528810)
            .verifyComplete();
    }
    
}

File: kirun-js/__tests__/engine/function/system/date/DifferenceOfTimestampsTest.ts
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';
import { KIRunFunctionRepository, KIRunSchemaRepository } from '../../../../../src';
import { DifferenceOfTimestamps } from '../../../../../src/engine/function/system/date/DifferenceOfTimestamps';

const differenceOfTimestamps: DifferenceOfTimestamps = new DifferenceOfTimestamps();

const fep: FunctionExecutionParameters = new FunctionExecutionParameters(
    new KIRunFunctionRepository(),
    new KIRunSchemaRepository()
);


describe('testing DifferenceOfTimestamps for invalid dates', () => {
    

    test('invalid date one', async () => {

        fep.setArguments(new Map([['isoDateOne', '2029-05-95T06:04:18.073Z'], ['isoDateTwo', '2029-05-05T06:04:18.073Z']]));

        await expect( () => differenceOfTimestamps.execute(fep)).rejects.toThrow();
    })

    test('invalid date one', async () => {

        fep.setArguments(new Map([['isoDateOne', '2029-05-05T06:04:18.073Z']]));

        await expect( () => differenceOfTimestamps.execute(fep)).rejects.toThrow();
    })
})

describe('testing DifferenceOfTimestamps for valid dates', () => {

    test('valid one', async () => {

        fep.setArguments(new Map([['isoDateOne', '2024-09-13T23:52:34.633-05:30'], ['isoDateTwo', '2024-09-13T23:52:34.633Z']]));

        expect((await differenceOfTimestamps.execute(fep)).allResults()[0].getResult().get('result')).toBe(-330);
    })

    test('valid two', async () => {

        fep.setArguments(new Map([['isoDateOne', '2024-09-13T23:52:34.633-05:30'], ['isoDateTwo', '2024-09-12T23:52:34.633Z']]));

        expect((await differenceOfTimestamps.execute(fep)).allResults()[0].getResult().get('result')).toBe(-1770);
    })

    test('valid three', async () => {

        fep.setArguments(new Map([['isoDateOne', '2023-09-12T23:52:34.633Z'], ['isoDateTwo', '2024-09-13T23:52:34.633-05:30']]));

        expect((await differenceOfTimestamps.execute(fep)).allResults()[0].getResult().get('result')).toBe(528810);
    })
})
